File: app/src/main/java/com/fuchsia/shotokanwkf/kataAdapter.java
package com.fuchsia.shotokanwkf;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;

import com.firebase.ui.database.FirebaseRecyclerAdapter;
import com.firebase.ui.database.FirebaseRecyclerOptions;
import com.google.android.gms.ads.FullScreenContentCallback;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;


public class kataAdapter extends FirebaseRecyclerAdapter<katamodel,kataAdapter.myviewholder> {


    public kataAdapter(FirebaseRecyclerOptions<katamodel> options) {
        super(options);
    }

    @Override
    protected void onBindViewHolder(@NonNull final myviewholder holder, final int position, @NonNull final katamodel katamodel)
    {
        holder.textView.setText(katamodel.getName());


        holder.vid.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View view) {

                AppCompatActivity activity= (AppCompatActivity) view.getContext();

                if (Admob.mInterstitialAd != null) {

                    Admob.mInterstitialAd.show(basicKarate.getInstance());

                    Admob.mInterstitialAd.setFullScreenContentCallback(new FullScreenContentCallback() {

                        @Override
                        public void onAdDismissedFullScreenContent() {

                            Intent intent =new Intent(activity, videoPlayer.class);
                            intent.putExtra("nam", katamodel.getURL());
                            activity.startActivity(intent);

                            Admob.mInterstitialAd = null;
                            Admob.loadInter(kataList.getInstance());

                        }
                    });
                }
                else{

                    Intent intent =new Intent(activity, videoPlayer.class);
                    intent.putExtra("nam", katamodel.getURL());
                    activity.startActivity(intent);

                    Admob.mInterstitialAd = null;
                    Admob.loadInter(kataList.getInstance());
                }


            }
        });

        holder.pic.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                AppCompatActivity activity= (AppCompatActivity) view.getContext();


                if (Admob.mInterstitialAd != null) {

                    Admob.mInterstitialAd.show(basicKarate.getInstance());

                    Admob.mInterstitialAd.setFullScreenContentCallback(new FullScreenContentCallback() {

                        @Override
                        public void onAdDismissedFullScreenContent() {

                            Intent intent =new Intent(activity, videoPlayer.class);
                            intent.putExtra("nam", katamodel.getURL());
                            activity.startActivity(intent);

                            Admob.mInterstitialAd = null;
                            Admob.loadInter(kataList.getInstance());

                        }
                    });
                }
                else{

                    Intent intent =new Intent(activity, videoPlayer.class);
                    intent.putExtra("nam", katamodel.getURL());
                    activity.startActivity(intent);

                    Admob.loadInter(kataList.getInstance());
                }


            }
        });

    }



    @NonNull
    @Override
    public myviewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType)
    {
        View view= LayoutInflater.from(parent.getContext()).inflate(R.layout.row,parent,false);
        return new myviewholder(view);
    }

    class myviewholder extends RecyclerView.ViewHolder
    {


        TextView textView;
        ImageView vid, pic;
        public myviewholder(@NonNull View itemView)
        {
            super(itemView);

            vid =itemView.findViewById(R.id.katavid);
            pic= itemView.findViewById(R.id.katapic);
            textView= itemView.findViewById ( R.id.kataname );


        }
    }


}

File: app/src/main/java/com/fuchsia/shotokanwkf/adapter/kataAdapter.kt
package com.fuchsia.shotokanwkf.adapter

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import com.firebase.ui.database.FirebaseRecyclerAdapter
import com.firebase.ui.database.FirebaseRecyclerOptions
import com.fuchsia.shotokanwkf.Admob
import com.fuchsia.shotokanwkf.Admob.loadInter
import com.fuchsia.shotokanwkf.R
import com.fuchsia.shotokanwkf.activity.kataList.Companion.instance
import com.fuchsia.shotokanwkf.model.katamodel
import com.fuchsia.shotokanwkf.activity.videoPlayer
import com.google.android.gms.ads.FullScreenContentCallback

class kataAdapter(options: FirebaseRecyclerOptions<katamodel?>?) :
    FirebaseRecyclerAdapter<katamodel, kataAdapter.myviewholder>(
        options!!
    ) {

    var context: Context? = null
    private lateinit var preferences: SharedPreferences
    private lateinit var editor: SharedPreferences.Editor

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): myviewholder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.katarowview, parent, false)
        return myviewholder(view)
    }
    override fun onBindViewHolder(holder: myviewholder, position: Int, katamodel: katamodel) {
        holder.textView.text = katamodel.name

        holder.vid.setOnClickListener { view ->

            val activity = view.context as AppCompatActivity
            preferences = activity.getSharedPreferences("subs", AppCompatActivity.MODE_PRIVATE)
            editor = preferences.edit()

            if (preferences.getBoolean("isPremium",true) ){
                val intent = Intent(activity, videoPlayer::class.java)
                intent.putExtra("nam", katamodel.url)
                activity.startActivity(intent)

            }else{
                val activity = view.context as AppCompatActivity
                if (Admob.mInterstitialAd != null) {
                    Admob.mInterstitialAd!!.show(instance!!)
                    Admob.mInterstitialAd!!.fullScreenContentCallback =
                        object : FullScreenContentCallback() {
                            override fun onAdDismissedFullScreenContent() {
                                val intent = Intent(activity, videoPlayer::class.java)
                                intent.putExtra("nam", katamodel.url)
                                activity.startActivity(intent)
                                Admob.mInterstitialAd = null
                                loadInter(instance!!)
                            }
                        }
                } else {
                    val intent = Intent(activity, videoPlayer::class.java)
                    intent.putExtra("nam", katamodel.url)
                    activity.startActivity(intent)
                    Admob.mInterstitialAd = null
                    loadInter(instance!!)
                }

            }

        }

    }

    inner class myviewholder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var textView: TextView
        var vid: ImageView

        init {
            vid = itemView.findViewById(R.id.katavid)
            textView = itemView.findViewById(R.id.kataname)
        }
    }
}